### PR TITLE
proxy: add non-abstract unix socket DAC properties

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: lxd/container.go [hunk 1]
```diff
@@ -458,6 +464,12 @@ func containerValidDevices(db *db.Cluster, devices types.Devices, profile bool,
 			if m["connect"] == "" {
 				return fmt.Errorf("Proxy device entry is missing the required \"connect\" property.")
 			}
+
+			if (!strings.HasPrefix(m["listen"], "unix:") ||
+				(strings.HasPrefix(m["listen"], "unix:") && strings.HasPrefix(m["listen"][len("unix:"):], "@"))) &&
```
Comment (stgraber) [inline lxd/container.go]: Wouldn't `strings.HasPrefix(m["listen"], "unix:@")` achieve the same here?

File: lxd/container.go [hunk 2]
```diff
@@ -458,6 +464,12 @@ func containerValidDevices(db *db.Cluster, devices types.Devices, profile bool,
...
+				(m["uid"] != "" || m["gid"] != "" || m["mode"] != "") {
+				return fmt.Errorf("Only proxy devices for non-abstract unix sockets can carry gid, uid, or mode properties")
```
Comment (stgraber) [inline lxd/container.go]: s/gid, uid/uid, gid/ (just a nit)

File: lxd/container.go [hunk 3]
```diff
@@ -227,11 +227,17 @@ func containerValidDeviceConfigKey(t, k string) bool {
 		}
 	case "proxy":
 		switch k {
-		case "listen":
```
Comment (stgraber) [inline lxd/container.go]: Can you sort those alphabetically (looks like you started to do that but then didn't do it all the way :))

File: lxd/main_forkproxy.go [hunk 4]
```diff
@@ -403,7 +403,7 @@ func (c *cmdForkproxy) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	// Sanity checks
-	if len(args) != 6 {
+	if len(args) < 6 || len(args) > 9 {
```
Comment (stgraber) [inline lxd/main_forkproxy.go]: Shouldn't that just be `!= 9` given that container_lxc.go always passes them (just setting them empty sometimes)?

File: lxd/container.go (hunk 1 above)
Comment (brauner) [reply lxd/container.go]: addressed

File: lxd/container.go (hunk 3 above)
Comment (brauner) [reply lxd/container.go]: Ah, i switched `connect` and `gid` accidently.

**Pull request merged**